load-code/pe/translate-characteristic-into-permission.yml

```
@@ -0,0 +1,22 @@
+rule:
+  meta:
+    name: translate characteristic into permission
```

what do you mean by "characteristic"?
```suggestion
    name: manipulate section memory permissions
```
maybe?

---

The characteristics of an image section header which can take values such as `IMAGE_SCN_MEM_READ`, `IMAGE_SCN_MEM_EXECUTE` and `IMAGE_SCN_MEM_WRITE`: https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-image_section_header

This rule is not to detect manipulation section memory permissions, but the translation of the section permissions specified in the image section header to page protection constants used in functions such as `CreateFileMapping` and `VirtualProtect`.

I am not sure how to make this clearer in the rule. Suggestions?

---

"characteristic" and "permission" are general terms. there's no indication from the rule name that we're talking about memory nor sections. im ok with keeping "translate" in there if we can clarify what types of characteristics and permissions we are talking about.

also, "translate" is an ok term, but its not great, since im not sure i've actually seen the phrase "translate memory characteristics" used before (though i agree it makes sense). "inspect section memory permissions" might also work.

---

I have updated to `inspect section memory permissions` in 9db6392b3803b6f1be988c4003af597620bc5f59. IS that ok @williballenthin?

---

👍🏼